lib/modules/platform/github/index.ts

```diff
@@ -739,8 +739,10 @@ export async function findPr({
 const REOPEN_THRESHOLD_MILLIS = 1000 * 60 * 60 * 24 * 7;
 
 async function ensureBranchSha(branchName: string, sha: string): Promise<void> {
-  const refUrl = `/repos/${config.repository}/git/refs/heads/${branchName}`;
+  const commitUrl = `/repos/${config.repository}/git/commits/${sha}`;
+  await githubApi.head(commitUrl, { memCache: false });
```

Comment:
Are you sure this addresses the right problem?

The problem I was addressing was that:
- branch didn't exist
- Renovate did a HEAD for the branch and got back a 200 OK so thinks it exists
- Renovate tries to patch a valid SHA to a non-existing ref

This PR seems to be addressing the (unreported) problem of if the SHA didn't exist

Reply:
I merged mine already